packages/vm/src/evm/evm.ts

```diff
@@ -60,6 +60,10 @@ export interface EVMOpts {
    * - `experimental`: behaviour can change on patch versions
    */
   common?: Common
+  /**
+   * A {@link StateManager} instance to use as the state store
+   */
+  stateManager: StateManager
```

Comment:
Just a note: this should rather be an in-between-step, after the EEI extraction (where external things like state access is encapsulated), a state manager dependency in the EVM itself should not be necessary any more.

Reply:
Yes, that describes it well.